components/net/cookie.rs

```diff
@@ -321,4 +343,226 @@ impl ServoCookie {
 
         true
     }
+
+    /// <https://www.ietf.org/archive/id/draft-ietf-httpbis-rfc6265bis-20.html#name-dates>
+    pub fn parse_date(string: &str) -> Option<OffsetDateTime> {
+        // NOTE: RFC6265 does not explicitly state that the value of the Expires attribute is
+        // case-insensitive, but it is a common behaviour in practice.
+        let lower_case_string = string.to_lowercase();
+        let string = lower_case_string.as_str();
+
+        // Step 1. Using the grammar below, divide the cookie-date into date-tokens.
+        // *OCTET
+        let any_octets = |input| Ok(("", input));
+        // delimiter = %x09 / %x20-2F / %x3B-40 / %x5B-60 / %x7B-7E
+        let delimiter: fn(&str) -> nom::IResult<&str, char> = |input| {
+            satisfy(
+                |c: char| matches!(c as u8,0x09 | 0x20..=0x2F | 0x3B..=0x40 | 0x5B..=0x60 | 0x7B..=0x7E),
```

Comment:
Iterating over chars and casting them to `u8` seems like it may result in some unexpected behavior. Unicode values will be truncated to just their lower 8bits. We probably should switch to using the nom's `bytes` matchers and convert the string `to_bytes`.